### DISCUSSION TITLE
Animation when changing rotation

### DISCUSSION THREAD

**Discussion by martabal** [willnguyen1312/zoom-image#246] [Q&A] [answered]
Hi !
Is it possible to animate the image when changing `currentRotation` value ?

**Comment by willnguyen1312** [comment, marked as answer]
Hi Martin, thanks for your question. When changing the `currentRotation` value, the library updates the [rotate](https://developer.mozilla.org/en-US/docs/Web/CSS/rotate) property of the image container. We can apply CSS transition on that element. Something like this should do it:

```css
transition: 0.5s;
```

https://github.com/willnguyen1312/zoom-image/assets/17794897/9d18aa1f-76a9-431b-af11-1ce517fdcb9b

I attached the demo link for your reference - https://willnguyen1312.github.io/zoom-image/examples/react.html

I hope it does answer your question. Thanks :)

**Comment by martabal** [comment]
Wow, thank you very much for your clear explanation and your reactivity !
